component/exporters/node/node_exporter.go

```diff
@@ -0,0 +1,97 @@
+package node_exporter
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	node_integration "github.com/grafana/agent/pkg/integrations/node_exporter"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "exporter.node",
+		Args:    node_integration.Config{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return NewComponent(opts, args.(node_integration.Config))
+		},
+	})
+}
+
+// Exports are simply a list of targets for a scraper to consume
+type Exports struct {
+	Targets []discovery.Target `river:"targets,attr"`
+}
+
+// Component for node exporter
+type Component struct {
+	log  log.Logger
+	opts component.Options
+
+	integration *node_integration.Integration
+
+	sync.Mutex
```

Comment:
nit: I'd prefer we avoid embedding sync.Mutex since it shows up at the public API level. 

It also makes it unclear _what_ the mutex is protecting, I prefer to group struct fields by the mutex that is protecting them to make it more clear to maintainers when the mutex is supposed to be locked:

```go
mut         sync.Mutex 
integration *node_integration.Integration
```